tests/test_it.py

```diff
@@ -140,6 +148,16 @@ def test_C402_fail_2():
     ]
 
 
+def test_C402_fail_3():
+    errors = run_flake8("""
+        lst = [('a', 1), ('b', 2), ('c', 3)]
+        dict(pair for pair in lst if pair[1] % 2 == 0)
```

Comment:
this can't be rewritten as a dict comprehension, because the node being iterated over is not a two-tuple, it's a variable that is bound to a two tuple

In general the approach of this change is back-to-front I think, it's blacklisting one particular kind of expression for dict comprehensions (`ast.Call`) from triggering the message, when we can just whitelist the one kind of expression that makes sense, a two-tuple.

I'm going to merge it as-is and follow up with a commit to change it to whitelisting